bonfire/processor.py

```diff
@@ -30,23 +30,82 @@ def _process_template(*args, **kwargs):
+    """
+    in_params = False
+    match = re.match(regex, value)
+    if match and match.groups()[0] in component_params:
+        in_params = True
+
+    log.debug(
+        "value '%s', regex r'%s', matches=%s, in params=%s", value, regex, bool(match), in_params
+    )
+
+    if match and in_params:
+        return True
+
+
+def _remove_untrusted_configs(data, params, path=None, current_dict=None, current_key=None):
+    """
+    Locate configurations within 'data' and remove them if not trusted.
+
+    Checks to see if any config matching a path listed in 'config.TRUSTED_PARAM_REGEX_FOR_PATH' is
+    found within 'data' dictionary and ensures the regex matches and that the parameter value is
+    set on the component's deploy config.
+    """
+    if not path:
+        path = ""
+
+    if isinstance(data, dict):
+        for key, value in copy.copy(data).items():
+            _remove_untrusted_configs(
+                value, params, path + f".{key}", current_dict=data, current_key=key
+            )
+
+    elif isinstance(data, list):
+        for index, value in enumerate(copy.copy(data)):
+            _remove_untrusted_configs(value, params, path + f"[{index}]")
+
+    # check if this value is at a path where we need to see a certain parameter in use
+    # in order to preserve it
+    for path_end, regex in conf.TRUSTED_REGEX_FOR_PATH.items():
+        # only check values if the path end is listed in TRUSTED_PARAM_REGEX_FOR_PATH
+        if not path.endswith(path_end):
             continue
 
-        removed = False
-        for d in i["spec"].get("deployments", []):
-            if "resources" in d["podSpec"]:
-                del d["podSpec"]["resources"]
-                removed = True
-        for p in i["spec"].get("pods", []):
-            if "resources" in p:
-                del p["resources"]
-                removed = True
+        if not _is_trusted_config(data, regex, params, path):
+            del current_dict[current_key]
+            log.debug("deleted untrusted config at '%s'", path)
 
-        if removed:
-            log.debug("removed resources from ClowdApp '%s'", i["metadata"]["name"])
+
+def _remove_untrusted_configs_for_template(template, params):
+    """
+    Removes untrusted configurations within the template.
+
+        A configuration is trusted if:
+        1. the value is defined using a template parameter
+        2. the parameter follows the proper syntax convention
+        3. the parameter is defined on the component in its deployment config
+
+    Checks template to see if any resources of kind listed in 'config.TRUSTED_CHECK_KINDS'
+    are found. If so, analyzes the config on those resources to see if any need to be removed.
+    """
+    for obj in template.get("objects"):
+        kind = obj.get("kind")
+        if kind not in conf.TRUSTED_CHECK_KINDS:
+            continue
+
+        log.debug("checking resources on %s '%s'", kind, obj["metadata"]["name"])
```

Comment:
Should you use `gets` here to make sure `obj["metadata"]["name"]` exists? I could picture a scenario where we actually bomb out in the log message